### PR TITLE
Enable ``numeric_only=False`` for ``DataFrame.count``

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files

Change itself is trivial, but sits on top of #10219

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -7918,7 +7926,13 @@ def repartition(df, divisions=None, force=False):
 def _reduction_chunk(x, aca_chunk=None, **kwargs):
     o = aca_chunk(x, **kwargs)
     # Return a dataframe so that the concatenated version is also a dataframe
-    return o.to_frame().T if is_series_like(o) else o
+    if is_series_like(o):
```
Comment (j-bennet) [inline dask/dataframe/core.py:7929]: What's going on here?

Comment (phofl) [reply dask/dataframe/core.py:7929]: This is from #10219.

When you have a DataFrame with ints/floats and bools, the reduction coerces to object, e.g. you get an object dtype series. We transpose to aggregate the chunks, which gives us an all object dtype DataFrame, which isn't what we want. So we have to infer the original dtypes, that's what I added here

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -7918,7 +7926,13 @@ def repartition(df, divisions=None, force=False):
...
+        result = o.to_frame().T
+        if o.dtype.kind == "O":
```
Comment (j-bennet) [inline dask/dataframe/core.py:7931]: Not a fun to multiple returns. If they were quick returns, maybe.

Comment (phofl) [reply dask/dataframe/core.py:7931]: Sorry I don't understand?

Comment (phofl) [reply dask/dataframe/core.py:7931]: These DataFrame have only one line btw, so this should be super fast

Comment (j-bennet) [reply dask/dataframe/core.py:7931]: Sorry, this comment was not very clear. It's just a code style nitpick.

Usually it's considered a good pattern to have a single point where a function returns control to its enclosing scope. Your lines 7929-7935 would look something like this:

```python
    if is_series_like(o):
        result = o.to_frame().T
        if o.dtype.kind == "O":
            result = result.infer_objects()
    else:
        result = o
    return result
```

Sometimes, this pattern leads to too much indentation. So people use "quick returns" to break out early, which makes the code more flat and easier to read:

```python
    if not is_series_like(o):
        return o
    result = o.to_frame().T
    if not o.dtype.kind == "O":
        return result
    return result.infer_objects()
```

Your snippet is following neither style. It has multiple points of returning control, but isn't saving any indentation levels.

I hope this helps.

Comment (phofl) [reply dask/dataframe/core.py:7931]: personally I don't like the first style. Imo this makes it harder to follow (recognising whether the result is modified again, especially if the code is more complex).

I'd be ok with the second one

File: dask/dataframe/core.py (hunk 1 above)
Comment (j-bennet) [reply dask/dataframe/core.py:7929]: I see what you're saying, but I'm not completely getting the use case, sorry. Would you have an example?

Comment (phofl) [reply dask/dataframe/core.py:7928]: The following example

```
df = pd.DataFrame(
    {
        "a": 1,
        "int": [1, 2],
        "float": [1.0, 2.0],
        "bool": True
    }
)
ddf = dd.from_pandas(df, npartitions=2)

x = ddf.min(numeric_only=True).compute()
```

runs into ``_reduction_chunk`` with
```
a           1
int         1
float     1.0
bool     True
dtype: object
```

Transposing the Series gives all object columns in the DataFrame which would translate to all object columns in the output. ``infer_objects`` casts the dtypes back.

File: dask/dataframe/core.py (hunk 2 above)
Comment (phofl) [reply dask/dataframe/core.py:7930]: Updated, also in the base PR

File: dask/dataframe/core.py (hunk 1 above)
Comment (j-bennet) [reply dask/dataframe/core.py]: 👍  thank you, that makes sense.

File: dask/dataframe/core.py [hunk 3]
```diff
@@ -2300,15 +2301,24 @@ def idxmin(self, axis=None, skipna=True, split_every=False):
                 result.divisions = (min(self.columns), max(self.columns))
             return result
 
-    @_numeric_only
     @derived_from(pd.DataFrame)
-    def count(self, axis=None, split_every=False, numeric_only=None):
+    def count(self, axis=None, split_every=False, numeric_only=False):
+        # This method is shared by DataFrame / Series, but only DataFrame
```
Comment (j-bennet) [inline dask/dataframe/core.py:2306]: Perhaps we should actually split the two methods, now that there is a difference in signature?

Comment (phofl) [reply dask/dataframe/core.py:2306]: Series actually has it' own method and calls this one.

**Review by j-bennet — APPROVED** [review]
👍

**Review by jrbourbeau — APPROVED** [review]
Thanks @phofl for the update and @j-bennet for reviewing

**Pull request merged**